src/plugins/vis_type_timeseries/public/application/components/vis_types/metric/vis.js

```diff
@@ -22,9 +26,14 @@ function getColors(props) {
   let background;
   if (model.background_color_rules) {
     model.background_color_rules.forEach((rule) => {
-      if (rule.operator && rule.value != null) {
-        const value = (series[0] && getLastValue(series[0].data)) || 0;
-        if (_[rule.operator](value, rule.value)) {
+      if (rule.operator) {
+        const value = getLastValue(series[0]?.data);
+        // This check is necessary for preventing from comparing null values with numeric rules.
+        const shouldOperate =
+          (isNull(rule.value) && OPERATORS_ALLOW_NULL[rule.operator]) ||
+          (!isNull(rule.value) && !isNull(value));
```

Comment:
It occurs in the code several times. Maybe we can create a separate 'check' function for this?

Reply:
Thanks for this suggestion) Sounds reasonable. Possible, even add OPERATORS and everything connected to them to the separate file. We will avoid duplicating code in metrics, top_n and gauge/vis.js files.